src/axom/CMakeLists.txt

```diff
@@ -10,17 +10,16 @@
 #------------------------------------------------------------------------------
 
 # Core is an essential part of Axom and cannot be turned off
-add_subdirectory(core)
-set(AXOM_ENABLE_CORE ON CACHE BOOL "" FORCE)
-set(AXOM_COMPONENTS_FULL core CACHE STRING "List of all components in Axom" FORCE)
-set(AXOM_COMPONENTS_ENABLED core
-    CACHE STRING "List of all enabled components in Axom" FORCE)
+axom_add_component(COMPONENT_NAME core DEFAULT_STATE ON)
+if(NOT AXOM_ENABLE_CORE)
```

Comment:
Curiosity question: if core is always required, why do we have the AXOM_ENABLE_CORE variable?

Reply:
Because it automatically created it =) I'll remove it.